types/event_bus.go

```diff
@@ -169,6 +169,15 @@ func (b *EventBus) PublishEventValidBlock(data EventDataRoundState) error {
 	return b.Publish(EventValidBlock, data)
 }
 
+func (b *EventBus) PublishEventPendingTx(data EventDataPendingTx) error {
+	// no explicit deadline for publishing events
```

Comment:
I don't like that this is blocking. Note that `notifyTxsAvailable` above is non-blocking. The consequence of potentially waiting for `eventBus` is slower transaction processing.